pkg/discovery/processor/cluster_processor.go

```diff
@@ -109,30 +120,32 @@ func checkNode(node *v1.Node, kc kubeclient.KubeHttpClientInterface) (float64, e
 }
 
 // Query the Kubernetes API Server to get the cluster nodes and namespaces and set in tge cluster object
-func (processor *ClusterProcessor) DiscoverCluster(kubeCluster *repository.KubeCluster) error {
+func (processor *ClusterProcessor) DiscoverCluster() (*repository.KubeCluster, error) {
 	// If connection to cluster is successful, then the cluster object is not null
-	if kubeCluster == nil {
-		return fmt.Errorf("Failed to connect to cluster")
+	if !processor.validationResult.IsValidated {
```

Comment:
The logical here is: **_If the cluster is not get validated, then won't perform discovery?_**

I think this could be a problem: If the initial validation has failed, then before the next validation request, even if the cluster is reachable, discovery requests will still fail?

Reply:
In addition,  If validation is failed, the `processor.validationResult` could be a `nil` pointer.